libraries/state_history/include/eosio/state_history/log.hpp

```diff
@@ -433,9 +435,9 @@ class state_history_log {
          detail::counter cnt;
          {
             bio::filtering_ostreambuf buf;
-            buf.push(boost::ref(cnt));
-            buf.push(bio::zlib_compressor(boost::iostreams::zlib::no_compression));
-            buf.push(bio::file_descriptor_sink(stream.fileno(), bio::never_close_handle));
+            buf.push(boost::ref(cnt), ship_log_iostreams_buffer_size);
+            buf.push(bio::zlib_compressor(bio::zlib::no_compression, ship_log_iostreams_buffer_size));
+            buf.push(bio::file_descriptor_sink(stream.fileno(), bio::never_close_handle), ship_log_iostreams_buffer_size);
```

Comment:
Will the buffer be flushed in case of crashes?

Reply:
Since the buffer is entirely owned by the `bio::filtering_istreambuf` it seems that the buffer would need to be emptied (really all buffers emptied and flushed through the pipeline) before destruction of `bio::filtering_istreambuf` otherwise data would be lost no matter crash or not.

I think this happens in the streambuf's close_impl(),
https://github.com/boostorg/iostreams/blob/develop/include/boost/iostreams/detail/streambuf/indirect_streambuf.hpp#L390
where sync() will then call sync_impl(),
https://github.com/boostorg/iostreams/blob/develop/include/boost/iostreams/detail/streambuf/indirect_streambuf.hpp#L410
which looks like it's writing out anything left in the 'put buffer'